packages/x-data-grid/src/components/menu/GridMenu.tsx

```diff
@@ -43,31 +42,24 @@ const useUtilityClasses = (ownerState: OwnerState) => {
   return composeClasses(slots, getDataGridUtilityClass, classes);
 };
 
-const GridMenuRoot = styled(Popper, {
+const GridMenuRoot = styled(NotRendered<GridSlotProps['basePopper']>, {
```

Comment:
I find this diff strange. What if we were to import https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Popper/BasePopper.tsx and style this one instead.

But even then, I'm unclear about the need. Maybe what confuses me is in the order in which we do this. Shouldn't we first move `GridMenuRoot` out of the unstyled version? Once we do, the fewer React component wrappers, the fastest the grid is.

Reply:
What we need to do here is wrap `rootProps.slots.basePopper`, but it's not available at initialization. I'm using this `NotRendered` component just to make the typechecking work. At runtime, `as={slots.basePopper}` needs to be used.

As for the styled/unstyled changes, I'm still evaluating how to do them.